src/go/rdctl/cmd/factoryReset.go

```diff
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
 		if err != nil {
 			return err
 		}
-		// Note that this command's only flag is default to not remove k8s cache
-		// but the server takes an optional flag meaning the opposite (as per issues 1701 and 2408)
-		factoryResetData.KeepSystemImages = !removeKubernetesCache
-		jsonBuffer, err := json.Marshal(factoryResetData)
+		cmd.SilenceUsage = true
+		_, err = doShutdown()
 		if err != nil {
 			return err
 		}
-		result, err := processRequestForUtility(doRequestWithPayload("PUT", versionCommand("", "factory_reset"), bytes.NewBuffer(jsonBuffer)))
-		if err != nil {
-			return err
-		}
-		fmt.Println(string(result))
-		return nil
+		return map[string]func() error{
+			"darwin":  deleteDarwinData,
+			"linux":   deleteLinuxData,
+			"windows": deleteWindowsData,
+		}[runtime.GOOS]()
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(factoryResetCmd)
 	factoryResetCmd.Flags().BoolVar(&removeKubernetesCache, "remove-kubernetes-cache", false, "If specified, also removes the cached Kubernetes images.")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.Verbose, "verbose", false, "Be verbose")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.DontWait, "dont-wait", false, "Don't wait for shutdown to be confirmed")
+}
+
+func deleteDarwinData() error {
+	libraryPath := path.Join(os.Getenv("HOME"), "Library")
+	appHomePath := path.Join(libraryPath, "Application Support", "rancher-desktop")
```

Comment:
`Application Support` is [`os.UserConfigDir()`](https://pkg.go.dev/os@go1.19.3#UserConfigDir).

Reply:
Continuing using standard os directory calls...